utils/math.hh

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+/*
+ * Copyright (C) 2023-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#include <cmath>
+
+namespace utils {
+
+template <unsigned i>
+constexpr double const_log() {
+    return std::log(i);
+}
+
+template <double x>
+constexpr double const_log() {
+    return std::log(x);
+}
+
+template <unsigned i>
+constexpr double inv_const_log() {
+    return 1.0 / std::log(i);
+}
+
+template <double x>
+constexpr double inv_const_log() {
+    return 1.0 / std::log(x);
+}
+
+template <unsigned base>
+inline double log_base(double x) {
+    return std::log(x) * inv_const_log<base>();
```

Comment:
There's no need for such optimization. The compiler knows how to compute the logarithm of a constant.

Reply:
Also, it's not helping to make the backlog tracker exception safe.